src/GraphBLAS-sharp.Backend/Quotes/Arithmetic.fs

```diff
@@ -36,6 +36,16 @@ module ArithmeticOperations =
 
             if res = zero then None else Some res @>
 
+    let inline mkNumericSumAsMul zero =
+        <@ fun (x: 't option) (y: 't option) ->
+            let mutable res = zero
+
+            match x, y with
+            | Some f, Some s -> res <- f + s
+            | _ -> ()
+
+            if res = zero then None else Some res @>
```

Comment:
Is it cirrect? Suppose, graph has negative edges. In this case one can get a path of length 0.

Reply:
This operation is used in SSSP and `Int32.MaxValue` is passed as zero, so if there is a path of length 0 it will be returned as `Some 0`

Reply:
What? Why `Int32.MaxValue` is used as zero? We have `None` for it.

Reply:
Returning non-unit value from match causes Brahma exception

Reply:
But in first case we pass `Int32.MaxValue` as zero and we compare result with it. So your code will return `Some 0` too. And, by the way, is it wrong?
And in your case it is forbidden to have path of length `Int32.MaxValue`.

Reply:
Yeah, I meant `Some Int32.MaxValue` will be returned, `Some 0` is fine. 
I guess there is nothing wrong with `Some Int32.MaxValue` too, so I will remove the check here